### PR TITLE
upgrade sets to use generic and add cilint

### PR DESCRIPTION
**Pull request by ycyaoxdu**
add golangci-lint and upgrade sets.String to sets.Set[string]

### CODE REVIEW

File: utils/work/v1/workbuilder/workbuilder_test.go [hunk 1]
```diff
@@ -105,16 +102,16 @@ func newFakeManifest(object runtime.Object) workapiv1.Manifest {
...
 
-func newFakeWorkClient(works []workapiv1.ManifestWork) client.Client {
-	s := scheme.Scheme
-	s.AddKnownTypes(workapiv1.GroupVersion, &workapiv1.ManifestWork{}, &workapiv1.ManifestWorkList{})
-	var objects []runtime.Object
-	for i := 0; i < len(works); i++ {
-		objects = append(objects, &works[i])
-	}
-	return fake.NewClientBuilder().WithScheme(s).WithRuntimeObjects(objects...).Build()
+// func newFakeWorkClient(works []workapiv1.ManifestWork) client.Client {
```
Comment (qiujian16) [inline utils/work/v1/workbuilder/workbuilder_test.go]: why we comment this?

Comment (ycyaoxdu) [reply utils/work/v1/workbuilder/workbuilder_test.go]: it is not used, not sure if we need to remove this function? @qiujian16

Comment (qiujian16) [reply utils/work/v1/workbuilder/workbuilder_test.go]: yes we should remove it if it is not used.

**Pull request merged**